### DISCUSSION TITLE
"Translate" function missing for only some posts

### DISCUSSION THREAD

**Discussion by M-Stenzel** [mastodon/mastodon#20998] [General]
Hi community,

I am an admin (version 4.0.2) who enabled the "Translate" feature using api.deepl.com, however, some languages seem not to be detected. In the attached picture you see the missing "Translate" button for Japanese. As you can see, the interface language is English, and the "Translate" feature is enabled for German.
I wonder how come? deepl.com supports Japanese.

Any hints?

![Translate](https://user-images.githubusercontent.com/76916440/202690196-d8a0a0f7-6d44-4abc-955b-027c86b0d3df.jpg)

**Comment by M-Stenzel** [comment]
This is an addendum:

Detection of Japanese does indeed work with other posts on the same server. So it is not the language by itself.
As an example! The orginal post (which lacks the "Translate" function) is to be found here:

https://mstdn.jp/@yui_tang/109364031408559371

To better understand the problem I copied & pasted the japanese text into the deepl.com web interface <https://www.deepl.com/translator> (which for sure uses the deepl api) and the text was correctly recognised as Japanese and was translated.

**Comment by veale** [comment]
Does the metadata of the post label it as Japanese?

**Comment by M-Stenzel** [comment]
This is a very good point, and indeed there are two entries that relate to locale/language, that is

"<html lang='**ja**'..."

and

"<script id="initial-state" type="application/json">{"meta":{"streaming_api_base_url":"wss://...","access_token":null,"locale":**JA**"","domain":"...","title"..."..."

In the post that lacks the "Translate" function it is set to en/EN, **but**
when I check with this post (as an example)

https://kolektiva.social/@MF36/109369907044282282

I find the following in the source:

"<html lang='**en**'..."

and

"<script id="initial-state" type="application/json">{"meta":{"streaming_api_base_url":"wss://...","access_token":null,"locale":**EN**"","domain":"...","title"..."..."

still I do see the "Translate" function (Spanish, as expected), although both the html languange, and the streaming_api_base_url define **English** as language/locale. 

I am puzzled.
